File: fs-core/src/main/java/xyz/srclab/common/convert/UnsupportedConvertException.java
package xyz.srclab.common.convert;

import java.lang.reflect.Type;

/**
 * Represents the converting is unsupported.
 *
 * @author fredsuvn
 */
public class UnsupportedConvertException extends RuntimeException {

    public UnsupportedConvertException() {
    }

    public UnsupportedConvertException(String message) {
        super(message);
    }

    public UnsupportedConvertException(String message, Throwable cause) {
        super(message, cause);
    }

    public UnsupportedConvertException(Throwable cause) {
        super(cause);
    }

    public UnsupportedConvertException(Type fromType, Type targetType) {
        this("Unsupported convert from " + fromType.getTypeName() + " to " + targetType.getTypeName() + ".");
    }
}

File: boat-core/src/main/kotlin/xyz/srclab/common/convert/UnsupportedConvertException.kt
package xyz.srclab.common.convert

import java.lang.reflect.Type

open class UnsupportedConvertException : RuntimeException {
    constructor(fromType: Type, toType: Type) : super("Unsupported convert: $fromType -> $toType")
    constructor(from: Any?, toType: Type) : super("Unsupported convert: ${from?.javaClass} -> $toType")
}